### PR TITLE
Feature/home

### PR DESCRIPTION
**Pull request by Mikazian**
## Description

Please include a summary of the change or which issue is fixed.
- Create Home page for login and registration
- Create button component

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Area of change

- [x] Frontend
- [ ] Backend
- [ ] Other: .....

## General checklist:

- [x] My code follows the style guidelines of this project
- [x] I ran `bun run format`/`bun run lint` before commit
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation where needed
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] I have updated `develop` and merged to my branch before submitting pull request
- [x] My pull request generate no conflicts with `develop` branch
- [ ] I requested code review from other team members

## Frontend checklist:

- [x] I followed guidelines for `HTML`/`JSX`/`Typescript`
- [x] My `Javascript` generate no new console errors
- [ ] I tested my code cross browsers
- [x] My slice is pixel perfect for both desktop and mobile according to design
- [ ] I conducted basic QA to assure all features are working
- [x] I tested responsive for mobile and tablet resolutions

## Backend checklist:

- [ ] I tested admin by manually adding content from zero
- [ ] I created easy to use admin experience which is self-explanatory
- [ ] I added description to admin fields in hard-to-understand areas
- [x] I followed guidelines for naming `Typescript` variables
- [ ] I conducted basic QA to assure all features are working

### CODE REVIEW

**Review by greegouze — APPROVED** [review]

File: packages/frontend/web/src/components/svg/Logo.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+export default function Logo() {
```
Comment (greegouze) [inline packages/frontend/web/src/components/svg/Logo.tsx:1]: On a le même nom de de composant 'Logo'

Comment (Mikazian) [reply packages/frontend/web/src/components/svg/Logo.tsx]: done

**Review by greegouze — APPROVED** [review]

**Review by ElBarto444 — DISMISSED** [review]
C'est good

File: package.json [hunk 2]
```diff
@@ -58,5 +58,8 @@
     "prettier": "^3.1.0",
     "prettier-plugin-tailwindcss": "^0.5.7",
     "typescript": "^5.3.2"
+  },
+  "dependencies": {
+    "framer-motion": "^10.16.5"
```
[the remark this answers is not in the document]
Comment (Mikazian) [reply package.json]: done, framer motion a été enlevé du package et ajouté dans le package du front-end/web

**Review by greegouze — APPROVED** [review]

**Review by ElBarto444 — DISMISSED** [review]
Ça m'a l'air bon

File: packages/frontend/web/src/components/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        damping: 10,
+        stiffness: 200,
+        duration: 0.1,
+      }}
+      className={`${
+        isOutline ? 'bg-transparent' : 'bg-primary'
+      } border-primary text-light-light w-full max-w-[500px] shrink-0 rounded-lg border px-2 py-3 text-xl`}
+    >
+      {/* Display the button text */}
+      {word}
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Button.tsx:36]: Pourquoi ne pas utiliser `children` ?

File: packages/frontend/web/src/components/home/RandomSentence.tsx [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  'Welcome to Tindev, where each match is an infinite loop of love without breaks.',
+  'Welcome to Tindev, where finding love is as simple as declaring a variable.',
+  "Swipe right if you're ready to share your cookies and create unforgettable memories on Tindev.",
+  'Welcome to Tindev, where dating is like HTML: simple but with infinite potential.',
+];
+
+export default function RandomSentence() {
+  // Select a random sentence from the array "sentences"
+  const randomSentence: string =
+    sentences[Math.floor(Math.random() * sentences.length)];
```
Comment (YDrogen) [inline packages/frontend/web/src/components/home/RandomSentence.tsx:30]: A sortir hors du component

File: packages/frontend/web/src/components/home/RandomSentence.tsx [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  'Looking for someone who can handle your exceptions?',
+  "Swipe right if you're ready to enter an infinite loop of love on Tindev, where each iteration is better than the last.",
+  'Welcome to Tindev, where each match is an SQL query to find the perfect partner.',
+  'Swipe right if you want someone who appreciates your CSS jokes. On Tindev, we believe that love is like code, it must be stylish.',
+  'On Tindev, each match is a promise to never leave us in callback.',
+  'Welcome to Tindev, where each match is an infinite loop of love without breaks.',
+  'Welcome to Tindev, where finding love is as simple as declaring a variable.',
+  "Swipe right if you're ready to share your cookies and create unforgettable memories on Tindev.",
+  'Welcome to Tindev, where dating is like HTML: simple but with infinite potential.',
+];
```
Comment (YDrogen) [inline packages/frontend/web/src/components/home/RandomSentence.tsx:25]: Peut être sorti dans un .json puis importé

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/home/RandomSentence.tsx (hunk 5 above)
Comment (Mikazian) [reply packages/frontend/web/src/components/home/RandomSentence.tsx:25]: done

File: packages/frontend/web/src/components/home/RandomSentence.tsx (hunk 4 above)
Comment (Mikazian) [reply packages/frontend/web/src/components/home/RandomSentence.tsx:30]: done

**Review by greegouze — APPROVED** [review]

File: packages/frontend/web/src/components/home/sentences.json
Comment (YDrogen) [inline packages/frontend/web/src/components/home/sentences.json:1]: ce n'est pas un component
`src/data/home-sentences.json` par exemple serait plus approprié

File: packages/frontend/web/tsconfig.json [hunk 6]
```diff
@@ -25,6 +25,6 @@
     "noUnusedParameters": true,
     "noFallthroughCasesInSwitch": true
   },
-  "include": ["src"],
+  "include": ["src/**/*", "src/**/*.json"],
```
Comment (YDrogen) [inline packages/frontend/web/tsconfig.json:28]: `src` ne suffit pas ?

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/Button.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import { motion } from 'framer-motion';
+
+export default function Button({
+  children,
+  type,
+  isOutline,
+}: {
+  readonly children: string;
```
Comment (YDrogen) [inline packages/frontend/web/src/components/Button.tsx:8]: `children` n'est pas censé être un string

**Review by YDrogen — CHANGES_REQUESTED** [review]

File: packages/frontend/web/src/components/home/sentences.json
Comment (Mikazian) [reply packages/frontend/web/src/components/home/sentences.json:1]: done

File: packages/frontend/web/tsconfig.json (hunk 6 above)
Comment (Mikazian) [reply packages/frontend/web/tsconfig.json:28]: update at "src"

File: packages/frontend/web/src/components/Button.tsx (hunk 7 above)
Comment (Mikazian) [reply packages/frontend/web/src/components/Button.tsx:8]: update at React.ReatNode

**Review by greegouze — DISMISSED** [review]

File: packages/frontend/web/src/components/home/sentences.json
Comment (YDrogen) [reply packages/frontend/web/src/components/home/sentences.json:1]: C'est toujours pas un component 😭

**Comment by Mikazian** [comment]
conflicts revolved !

**Review by ElBarto444 — DISMISSED** [review]
C'est bon pour moi

File: packages/frontend/web/src/components/Button.tsx
Comment (YDrogen) [inline packages/frontend/web/src/components/Button.tsx:1]: Les balises natives HTML (`button` par exemple) ont des props typées `HTMLButtonElement` ou un truc du genre
Du coup tu peux extends ces props pour garder celles de bases et ajouter les tiennes
```typescript
interface ButtonProps extends HTMLButtonElement {
  readonly isOutline: boolean;
}
```
Ça évite de typer à la main `children` ou `type` dans ton cas ici

**Review by YDrogen — CHANGES_REQUESTED** [review]

**Review by greegouze — APPROVED** [review]

**Review by ElBarto444 — DISMISSED** [review]
c'est ok pour moi

**Review by YDrogen — CHANGES_REQUESTED** [review]
Déso, t'as oublié une conversation non résolue au dessus :(

**Comment by Mikazian** [comment]
J'ai déplacé le dossier data dans src

**Review by greegouze — APPROVED** [review]

**Review by ElBarto444 — APPROVED** [review]
C'est validé

**Review by YDrogen — APPROVED** [review]

**Pull request merged**